Code/GraphMol/CIPLabeler/Digraph.cpp

```diff
@@ -0,0 +1,232 @@
+//
+//
+//  Copyright (C) 2020 Schrödinger, LLC
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+
+#include <sstream>
+
+#include "Digraph.h"
+#include "CIPMol.h"
+#include "Node.h"
+#include "Edge.h"
+
+namespace RDKit {
+namespace CIPLabeler {
+
+namespace {
+
+/**
+ * Upper limit on the size of the digraph, stops out of memory error with a
+ * more graceful failure. 0=Infinite
+ */
+const int MAX_NODE_COUNT = 100000;
+
+/**
+ * Used for debugging only, 0=Infinite
+ */
+const int MAX_NODE_DIST = 0;
+} // namespace
+
+void Digraph::addEdge(Node *beg, Bond *bond, Node *end) {
+  auto e = new Edge(beg, end, bond);
+  beg->add(e);
```

Comment:
probably not a bad idea to add a comment here pointing out that this takes ownership of `e`

Reply:
This will change soon. We are planning a change such that the Digraph owns all nodes and edges.